relayer/chains/processor/chain_processor.go

```diff
@@ -0,0 +1,24 @@
+package processor
+
+import (
+	"context"
+)
+
+// ChainProcessor interface is reponsible for polling blocks and emitting ibc message events to the PathProcessors
+// it is also responsible for tracking open channels and not sending messages to the PathProcessors for closed channels
+type ChainProcessor interface {
+	// starts the query loop for the chain which will gather applicable ibc messages and push events out to the relevant PathProcessors
```

Comment:
This comment does not explain what `initialBlockHistory` or `errCh` do, and it's not obvious from the method name either.